<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
    <script>
        fetch("https://jsonplaceholder.typicode.com/posts")
            .then(function(res){
                return res.json();
            })
            .then(function(posts){
                console.log(posts);
            })

        fetch("https://jsonplaceholder.typicode.com/posts")
            .then(function(res){
                console.log("First then!");
            },
            //  function(err){
            //     console.log("Error occurred:", err);
            // }
        )
            .then(function(res){
                console.log("Second then !");
                throw new Error("Error in second then");
                //if a error is thrown in a then, it will be caught by the next catch and the chaining will stop
            }
            // , function(err){
            //     console.log("Error occurred:", err);
            // }
        )
            .then(function(res){
                console.log("Third then !");
            }
            // , function(err){ //local catch method, use when the method is not chained
            //     //this will not break the chain, but will catch the error in this then
            //     console.log("Error occurred:", err);
            // }
        )
            .then(function(res){
                console.log("Fourth then !");
            })
            // global catch method- this will break the chain
            .catch(function(err){
                console.log("Error occurred:", err);
            });
    </script>
    </script>
</head>
<body>
    
</body>
</html>